appserver/payara-appserver-modules/microprofile/opentracing/src/main/java/fish/payara/microprofile/opentracing/jaxrs/JaxrsContainerRequestTracingExceptionMapper.java

```diff
@@ -44,31 +44,38 @@
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.Response.StatusType;
 import javax.ws.rs.ext.ExceptionMapper;
+import java.net.URI;
 
 /**
  * ExceptionMapper that catches all Exceptions. We need this because we need to add details about any exceptions to the
  * active span - if this isn't here, we don't go back through the container filter.
- * 
+ *
  * @author Andrew Pielage <andrew.pielage@payara.fish>
  */
 @Priority(Integer.MAX_VALUE)
 public class JaxrsContainerRequestTracingExceptionMapper implements ExceptionMapper<Throwable> {
 
     @Override
-    public Response toResponse(Throwable exception) {      
+    public Response toResponse(Throwable exception) {
         StatusType status = Response.Status.INTERNAL_SERVER_ERROR;
-        
-        // Get the status if available
+        URI location = null;
+
+        // Get the status and location if available
         if (exception instanceof WebApplicationException) {
-            status = ((WebApplicationException) exception).getResponse().getStatusInfo();
-        }    
-        
+            Response response = ((WebApplicationException) exception).getResponse();
+            location = response.getLocation();
+            status = response.getStatusInfo();
+        }
+
+        Response.ResponseBuilder responseBuilder = Response.status(status);
```

Comment:
Why is a new response being created here at all? Shouldn't the WebApplicationException response be used as a base if it exists?

Reply:
Already merged into master, raised issue PAYARA-3262 to track.